liquid-rust-core/src/resolve.rs

```diff
@@ -59,6 +61,66 @@ impl<'tcx> Resolver<'tcx> {
         Ok(Resolver { tcx, diagnostics, name_res_table, parent })
     }
 
+    /// `desugar_def_sig(sig)` translates a `DefFnSig` into a `core::FnSig`, and is used by
+    ///  * functions with specs whose `BareFnSig` is first resolved by `resolve_bare_sig` and
+    ///  * functions without specs whose `rust::FnSig` is used to generate a default `DefFnSig`
+    fn desugar(
+        &mut self,
+        def_id: LocalDefId,
+        dsig: surface::DefFnSig,
+    ) -> Result<ty::FnSig, ErrorReported> {
+        let mut subst = self.resolve_rust_generics(def_id);
+        let name_gen = IndexGen::new();
+        desugar(dsig, &name_gen, &mut subst, &mut self.diagnostics)
+    }
+
+    /// `default_fn_sig(f, span)` uses the type-context to generate a default `DefFnSig` for `f`
+    fn default_fn_sig(
```

Comment:
I don't think this function belongs in `Resolver`. This could be moved to `surface::default_fn_sig` which could have the signature

```rust
(TyCtxt, LocalDefId, Span) -> DefFnSig
```